sdk/formrecognizer/azure-ai-formrecognizer/src/FormRecognizer.md

```diff
@@ -0,0 +1,463 @@
+    public String/DateTime transactionTime;
+}
+
+public class ReceiptItem {
+    public String name;
+    public int quantity;
+    public double totalPrice;
+}
+
+public enum ReceiptType {
+    Unrecognized;
+    Itemized;
+}
+
+public class RawReceiptExtraction {
+    public List<RawReceiptItemExtraction> items;
+    public PredefinedField merchantAddress;
+    public PredefinedField merchantName;
+    public PredefinedField merchantPhoneNumber;
+    public PredefinedField subtotal;
+    public PredefinedField tax;
+    public PredefinedField tip;
+    public PredefinedField total;
+    public PredefinedField transactionDate;
+    public PredefinedField transactionTime;
+}
+
+public class RawReceiptItemExtraction {
+    public PredefinedField name;
+    public PredefinedField quantity;
+    public PredefinedField totalPrice;
+}
+
+ public class PredefinedField {
+    public double confidence;
+    public String name;
+    public ValueType type;
+    public ExtractedText extractedValue;
+}
+public class LabeledFieldForm {
+    public String formType;
+    public PageRange PageRange;
+    public List<LabeledPageInfo> pages;
+}
+public class LabeledPageInfo {
+    public List<PredefinedField> fields;
+    public int PageNumber;
+    public List<ExtractedTable> Tables;
+}
+
+public enum ValueType {
+    StringType,
+    Date,
+    Time,
+    PhoneNumber,
+    Number,
+    IntegerType,
+    Array,
+    ObjectType,
+}
+
+// Analyze Layout
+public class ExtractedLayoutResult {
+    private List<LayoutPage> pages;
+    private List<RawPageExtraction> rawPageExtractions;
+}
+
+public class LayoutPage {
+    private int pageNumber;
+    private List<ExtractedTable> tables;
+}
+
+public class ExtractedLine extends ExtractedText {
+    private TextLanguage language;
+    private List<ExtractedWord> words;
+}
+
+public enum TrainingStatus {
+    SUCCEEDED("succeeded"),
+    PARTIALLY_SUCCEEDED("partiallySucceeded"),
+    FAILED("failed");
+}
+
+public enum ModelStatus {
+    CREATING("creating"),
+    READY("ready"),
+    INVALID("invalid");
+}
+```
+
+## Samples
+### [Unsupervised] Train and analyze with custom model
+Train without labels
+- By default, Form Recognizer uses unsupervised learning to understand the layout and relationships between fields and entries in your forms.
+
+```java
+String trainingSetSource = "<storage-sas-url>";
+PollerFlux<OperationResult, TrainedModel> trainingPoller = client.beginModelTraining(trainingSetSource);
+
+TrainedModel unsupervisedModel = trainingPoller
+    .last()
+    .flatMap(trainingOperationResponse -> {
+        if (trainingOperationResponse.getStatus().isComplete()) {
+            // training completed successfully, retrieving final result.
+            return trainingOperationResponse.getFinalResult();
+        } else {
+            throw new RuntimeException("Polling completed unsuccessfully with status:"
+                + trainingOperationResponse.getStatus());
+        }
+    }).block();
```

Comment:
In terms of sample code, it would be best to demonstrate a proper async example that doesn't block, or else, uses the sync client and operates as appropriate for there. Right now we are in the worst of both worlds.